src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/EC/EccTestBase.cs

```diff
@@ -0,0 +1,235 @@
+using System.Linq;
+using System.Security.Cryptography.Tests;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    /// <summary>
+    /// Input and helper methods for EC classes
+    /// </summary>
+    public abstract class EccTestBase
+    {
+#if netcoreapp
+        internal const string ECDSA_P224_OID_VALUE = "1.3.132.0.33"; // Also called nistP224 or secP224r1
+        internal const string ECDSA_P256_OID_VALUE = "1.2.840.10045.3.1.7"; // Also called nistP256, secP256r1 or prime256v1(OpenSsl)
+        internal const string ECDSA_P384_OID_VALUE = "1.3.132.0.34"; // Also called nistP384 or secP384r1
+        internal const string ECDSA_P521_OID_VALUE = "1.3.132.0.35"; // Also called nistP521 or secP521r1
+        internal const string ECDSA_Sect193r1_OID_VALUE = "1.3.132.0.24"; //Char-2 curve
+
+        public static IEnumerable<object[]> TestCurvesFull
+        {
+            get
+            {
+                var curveDefs =
+                    from curveDef in TestCurvesRaw
+                    where curveDef.IsCurveValidOnPlatform == true
+                    select curveDef;
+
+                foreach (CurveDef cd in curveDefs)
+                    yield return new[] { cd };
+
+                // return again with IncludePrivate = true
+                foreach (CurveDef cd in curveDefs)
+                {
+                    cd.IncludePrivate = true;
+                    yield return new[] { cd };
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> TestCurves
+        {
+            get
+            {
+                var curveDefs =
+                    from curveDef in TestCurvesRaw
+                    where curveDef.IsCurveValidOnPlatform == true
+                    select curveDef;
+
+                foreach (CurveDef curveDef in curveDefs)
+                    yield return new[] { curveDef };
+            }
+        }
+
+        public static IEnumerable<object[]> TestInvalidCurves
+        {
+            get
+            {
+                var curveDefs =
+                    from curveDef in TestCurvesRaw
+                    where curveDef.IsCurveValidOnPlatform == false
+                    select curveDef;
+
+                foreach (CurveDef curveDef in curveDefs)
+                    yield return new[] { curveDef };
+            }
+        }
+
+        public static IEnumerable<object[]> TestNewCurves
+        {
+            get
+            {
+                var curveDefs =
+                    from curveDef in TestCurvesRaw
+                    where
+                        curveDef.IsCurveValidOnPlatform == true &&
+                        curveDef.RequiredOnPlatform == false
+                    select curveDef;
+
+                foreach (CurveDef curveDef in curveDefs)
+                    yield return new[] { curveDef };
+            }
+        }
+
+        private static IEnumerable<CurveDef> TestCurvesRaw
```

Comment:
How did you choose the elliptic curve domain parameters to test?

Reply:
This was just a move from the ECDSA tests to be shared across the two ECC algorithms.  It looks mostly like it's a curve from each family, plus the three supported on Win7 and macOS.

Is there something specific you're looking for?

Reply:
I mostly want to make sure we're covering all common cases such as those listed in the specs on http://www.secg.org/